### PR TITLE
fix: don't check the podman default connection while activating the e…

### LINKED ISSUE #9124: Podman extension should be initialized async
### Bug description

While testing a PR on my machine, I noticed the Podman extension does not start anymore because it now takes more than 20s.
It should be made async

### Operating system

Win11Pro

### Installation Method

None

### Version

1.11.1

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### Additional context

_No response_

### LINKED ISSUE #9576: Podman Desktop 1.13.2 - Podman Extension (v1.13.0-next) Fails to Start
### Bug description

I've recently updated to Podman Desktop 1.13.2, and since the update, the Podman extension (v1.13.0-next) fails to start, rendering the application unusable. Downgrading to Podman Desktop v1.12.0 resolves the issue and everything works as expected.

A colleague of mine has experienced the same behavior.

### Operating system

Windows Version 23H2

### Installation Method

Winget (Windows)

### Version

1.13.0

### Steps to reproduce

1. Update to version 1.13.2

### Relevant log output

Error: Extension podman-desktop.podman activation timed out after 20 seconds
Stack trace:
Error: Extension podman-desktop.podman activation timed out after 20 seconds
    at Timeout._onTimeout (C:\Users\XXX\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:151:67168)
    at listOnTimeout (node:internal/timers:581:17)
    at process.processTimers (node:internal/timers:519:7)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jeffmaury**
…xtension

Fixes #9124
Fixes #9576

### What does this PR do?

Prevent checking the default connection during Podman extension activation as it throws a dialog

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

#9124 

### How to test this PR?

1. Ensure the default connection is not aligned with the machine
2. Start Podman Desktop

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by axel7083 — COMMENTED** [review]
I just checkout the branch, stop my default machine, start another machine (windows). Then I started podman desktop using pnpm watch.

When the application boot, I am getting directly the following

image

And I do not have any failure on the podman extension side

image

Is this the expected behaviour ?

**Comment by jeffmaury** [comment]
> I just checkout the branch, stop my default machine, start another machine (windows). Then I started podman desktop using pnpm watch.
> 
> When the application boot, I am getting directly the following
> 
> image
> 
> And I do not have any failure on the podman extension side
> 
> image
> 
> Is this the expected behaviour ?

YES

**Review by axel7083 — APPROVED** [review]
LGTM 🚀 nice fix 👍 !

File: extensions/podman/packages/extension/src/extension.ts [hunk 1]
```diff
@@ -185,6 +185,7 @@ function notifySetupPodman(): void {
 export async function updateMachines(
   provider: extensionApi.Provider,
   podmanConfiguration: PodmanConfiguration,
+  checkDefaultConnection: boolean,
```
Comment (benoitf) [inline extensions/podman/packages/extension/src/extension.ts]: question: make it optional and default to false ?

so we would only one change in the codebase where we set true

Comment (jeffmaury) [reply extensions/podman/packages/extension/src/extension.ts]: I am on old Java developer 🥲 

Should be good now

**Review by benoitf — APPROVED** [review]

**Review by benoitf — APPROVED** [review]
suggestion: might miss the unit test to check if we call or not the method based on the arg

**Comment by benoitf** [comment]
rebasing the PR

**Comment by jeffmaury** [comment]
Closing as not required anymore since #10119

**Pull request closed without merging**